File: C/Queue/Queue_01.c
/*
===============================================================================
Program Name    : Queue Using Singly Linear Linked List
Program Number  : 01
File Name       : Queue01.c

Description     : This program contains the basic skeleton
                  of Queue Data Structure using Singly
                  Linear Linked List.

Date            : August 2026
Language        : C
Author          : Atharv Tushar Bhosale
===============================================================================
*/

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;

struct Queue
{
    PNODE first;
    int iCount;
};

typedef struct Queue QUEUE;

int main()
{
    QUEUE qobj;

    qobj.first = NULL;
    qobj.iCount = 0;

    return 0;
}

/*
===============================================================================
End of Program

Queue Using Singly Linear Linked List

Author  : Atharv Tushar Bhosale
Date    : August 2026
===============================================================================
*/

File: Java/Queue/Queue_01.java
/*
===============================================================================
Program Name    : Queue Skeleton
Program Number  : 001
File Name       : Queue_01.java

Description     : This program demonstrates the basic
                  structure of Queue using Singly
                  Linear Linked List.

Concepts Used:
1. Class and Object
2. Linked List
3. Queue Data Structure

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class node
{
    public int data;
    public node next;
}

class Queue
{
    private node first;
    private int iCount;

    public Queue()
    {
        first = null;
        iCount = 0;
    }
}

class Queue_01
{
    public static void main(String A[])
    {
        Queue qobj = new Queue();

        System.out.println("Queue created successfully");
    }
}

/*
===============================================================================
Sample Output

Queue created successfully

===============================================================================
End of Program
===============================================================================
*/
